### DISCUSSION TITLE
Opening SQLITE database connection wants to overwrite file.

### DISCUSSION THREAD

**Discussion by mark-hyde** [dbeaver/dbeaver#15715] [Q&A]
#### System information: 
- Operating system (distribution) and version
$ cat /etc/redhat-release 
Red Hat Enterprise Linux release 8.5 (Ootpa)

- DBeaver version
DBeaver Community                                   io.dbeaver.DBeaverCommunity                                      21.3.5                            stable              system
(Also checked 21.3.4)

- Additional extensions

#### Connection specification:
- Database name and version
SQLite version 3.26.0 2018-12-01 12:34:55
- Driver name
Sqlite
- Do you use tunnels or proxies (SSH, SOCKS, etc)? 
No
#### Describe the problem you're observing:
I am not able to easily open a new connection to an existing sqlite database file.

#### Steps to reproduce, if exist:
In flatpak DBeaver Community versions 21.2.5, (which works!) I am able to open an existing sqlite db by creating a connection:
(new connection icon->sqlite->"Browse"->use the 'Choose database file' dialog to select the database-> Click 'Open')

However in the latest flatpak version 21.3.5 (and at least one other previous 21.3.x ) this is not possible because the 'Choose database file' dialog has a 'Save' button rather than an 'Open' button. 
If you click 'Save' then it asks if you want to overwrite the existing file (which we don't!!)

Workarounds are: 
A - change the name of the file in the dialog from 'dummy.db' to 'dummy.dbx', save the  connection, and then edit the connection to point at the original file.
B - Rollback the version to 21.2.5

**Comment by ShadelessFox** [comment]
Hello @mark-hyde,

Pressing `Save` does not overwrite the existing file.

We had to make this dialog saveable to allow specifying non-existing files so that the database could create one.

**Comment by mark-hyde** [comment]
Hi @ShadelessFox,
Thanks for the info, that's good to know.
However the dialogue box when you click 'Save' explicitly says it will overwrite it! 
I didn't go so far as to test that as I didn't want to lose my data.

Would there be any way to make that more clear in the interface (so as not to scare those of us who don't know different!)?

**Reply by ShadelessFox** [reply]
The behavior you are describing is actually a platform-specific. For example, on Windows, it doesn't say anything about overwriting files.

The title of this dialog already says that you need to either select an existing database file or create a new one. Maybe we can rephrase it somehow.

**Comment by mark-hyde** [comment]
I upgraded my version to the latest in flatpak to test it out (on an unimportant file this time! 'cache/cache.db' ).
The specific verbiage on my platform is: 
'
A file named "cache.db" already exists. Do you want to replace it?
The file already exists in "cache". Replacing it will overwrite its contents.
'
I've tried to paste a screenshot here...
![image](https://user-images.githubusercontent.com/9009236/156818431-f9517d0b-fa54-4658-888e-105cbbf3acd9.png)

I can confirm that when I click 'Replace' it opens the original file so all good in terms of what it does, however quite scary without the developer telling you it's OK!

I wonder if 'Creating a new connection to an existing database' and 'Creating a new Database (and then a connection to it)' are conceptually 2 different things?

**Reply by ShadelessFox** [reply]
I'm unsure how we could make it user-friendly. Some custom file pickers offer a button for directly creating a new file, but in the case of a platform-specific file picker, it won't work.

Yes, these are different things. For example, a database (an SQLite one) represents data, whether a connection is just a bridge between DBeaver and that data. Even though you have multiple bridges, you will eventually get to the same place.

**Reply by mark-hyde** [reply]
Rereading my post makes me sound a bit critical sorry - I'd like to say that I don't use dbeaver very much on a day to day basis, but when I do it's incredibly useful so I'd like to thank you for all your hard work.

From my point of view - I know what to do now so the issue is 'solved' for me personally. 

But in the interest of discussion: 
The previous  mechanism 'Opened' a connection to an existing sqlite database file, and the dialogue said 'open' and it was user-friendly and unambiguous.
The new mechanism now (in addition) allows you to create a new database file and then create a 'connection' to it in addition to the previous functionality. But the downside is that the dialogue has you click a 'save' button rather than 'open' and gives a warning that you are going to overwrite a file (even if you are not in reality going to do so).
I'm not familiar enough with Dbeaver when working with other databases, but is there similar functionality when creating a connection to a mysql database instance? (i.e is it possible, when creating a connection to a database instance, to also create a new database schema within the instance?)
My view (as mainly an observer - and so please ignore me if you like!) is that the 'open connection' functionality should not also be a 'create database' functionality? Perhaps the 'create new sqlite database file' should be somewhere else in the interface?

**Comment by mark-hyde** [comment]
(For reference looks like this was where it was introduced - https://github.com/dbeaver/dbeaver/issues/14350 )

**Reply by ShadelessFox** [reply]
Sorry for the late response.

Can you please create a new issue for improving this behavior?

You can reference this discussion in it.

**Reply by ShadelessFox** [reply]
Somebody already has created an issue: #15827

**Reply by mark-hyde** [reply]
OK Thanks, I've subscribed to updates from it

**Comment by davidhooey** [comment]
What about having a Create New Database and Select Existing Database as opposed to a generic Browse button? This will allow for the correct dialog to be shown based on the user's action.
